src/compiler/checker.ts

```diff
@@ -28342,6 +28342,9 @@ namespace ts {
                             if (flags & ModifierFlags.Private) {
                                 return grammarErrorOnNode(modifier, Diagnostics._0_modifier_cannot_be_used_with_1_modifier, "private", "abstract");
                             }
+                            if (flags & ModifierFlags.Async) {
+                                return grammarErrorOnNode(modifier, Diagnostics._0_modifier_cannot_be_used_with_1_modifier, "async", "abstract");

```

Comment:
I think the error spans that we give are on the wrong keyword in all of these places. For example, the problem isn't that you've declared the thing as `abstract`, it's that it makes no sense to mark an abstract member as `async`.


But I guess that's not your problem. It *might* be suitable for a follow-up PR if it's not too disruptive.

Reply:
Do you mean the error should appear on `async` instead? I agree that's not really in the scope of *this* PR but that is one I would be happy to tackle as an experiment to evaluate whether it is too disruptive.